lib/collector.js

```diff
@@ -25,10 +27,24 @@ class Collector {
       this.#timer = null;
     }
     if (msec === 0) return;
-    const handler = () => this.fail(new Error('Collector timed out'));
+    const handler = () => {
+      if (Object.keys(this.defaultValue).length === 0) {
+        return void this.fail(new Error('Collector timed out'));
+      }
+      if (Object.keys(this.data).length === 0) this.#default();
+    };
     this.#timer = setTimeout(handler, msec);
   }
 
+  #default() {
+    for (const [key, value] of Object.entries(this.defaultValue)) {
+      this.set(key, value);
+    }
+    if (!this.done) {
+      return void this.fail(new Error('Not enough default values'));
```

Comment:
@zoff4ick Why should we do this? We throw this error in the fail function, what's the difference doing this in the `#timeout` or `#default`? I think it's just unnecessary since this error belong to the `#default` and not to `#timeout`

Reply:
I think in this case we shouldn't use default values at all because as I understand this, every user must have `cryptoSignatur`, so user cant't exist without this value. Then why should we set default values? I mean, that for those cases with dynamic created values we shouldn't use default values and then we become error from timer(time is over). 
I though, we seemed to discuss this too at the meeting yesterday.

Reply:
Okay, I agree with you. 

But I think now we don't need catch this error in timer or generate this error in timer. It's not timers error, this error must belong to the `#default` and only to default values. So I suggest renaming this error('Not enough default values') to something more wide. Like `After a time and all default values were used, not all keys were filled`.  I hope those error will be quite understandable and will not shock in such case as you callled. We can discuss how best to rename this error, but I think it's best to just rename it and not involve the timer in this story.  

And yes, after this error you can still use your collector further.

Reply:
I think we can discuss this on the next meeting. But I think we shouldn't send the same error for timer(when there are no default values) and for default values(when we use default values). In your suggestion we use alway default values and timer loses his functionality completely.
So handler should look something like this:

`
const handler = () => {
    if (Object.keys(this.defaultValue).length !== 0) this.#default(); 
    if (!this.done) return void this.fail(new Error('Collector timed out')); 
};`

**But we still get the same error as after timer and after applied the default values.** I find this the biggest problem, because letter questions may arise about what happened(For example when we use 2 collector in application and only one of them has default values, however they will return the same error).
So I think the best solution is to rename the error in `#default` function to something more large and wide(one of possible variants is above in my last comment or maybe something like `time is over and default values are used`). And so we can also remove this strange check/condition in handler, I mean `if (!this.done)`. Because this timer must work only when done is false, so it looks a little bit strange.

Reply:
@zoff4ick 
Now, after new commits in metautil, default values work like you wonted (after using default values you will get a timeout error). But i still think we need to discuss it one more time in meeting.